test/test.spec.js

```diff
@@ -32,3 +37,48 @@ test('removes custom prefix', async t => {
     t.false(f.name.includes('fixtures'))
   })
 })
+
+test('allows read of more files than ulimit maxfiles', async t => {
+  /** @type {string} */
+  let dir
+  try {
+    const totalFiles = 256
+    dir = await generateTestData(totalFiles)
+
+    // Restrict open files to less than the total files we'll read.
+    unlimited(totalFiles - 1)
+
+    const files = await getFilesFromPath(dir)
+    t.is(files.length, totalFiles)
+
+    // Restrict open files to less than the total files we'll read.
+    unlimited(totalFiles - 1)
```

Comment:
why call `unlimited` again after `getFilesFromPath`

Reply:
Make sure it really happened 🤷‍♂️